appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy

```diff
@@ -248,6 +274,12 @@ abstract class AbstractAppServiceIntegrationTest extends Specification
 		assert dayActivityForGoal._links.self == null  // This is not a top level or embedded resource
 	}
 
+	private Goal findWeekGoal(expectedValues, activityCategoryUrl)
+	{
+		def expectedValuesEndOfWeek = (expectedValues.size() == 1) ? expectedValues[getCurrentShortDay()] : expectedValues["Sat"]
```

Comment:
Why end of week? I thought the `WeekActivity` was constructed when the first `DayActivity` was constructed for that week?